### PR TITLE
Fix API expectations

### PR DESCRIPTION
**Pull request by vlm**
According to [revamp](https://github.com/sustrik/libdill/blob/master/rfc/bsd-socket-api-revamp.md#protocol-termination):

> Function “detach” MUST return the handle of the underlying protocol in case of success.

### CODE REVIEW

**Comment by sustrik** [comment]
The patch looks good. There's one more thing to check though: After detach have been called, all the pending data from TLS protocol must be cleaned up and the underlying socket can be used to run a different protocol. I am not familiar with OpenSSL, so it's hard for me to tell whether that's the case. btls_wait_close() looks a bit suspicious though. Can you have a look?

**Comment by vlm** [comment]
@sustrik , sorry I am missing what are you trying to say here. Could you rephrase?

**Comment by sustrik** [comment]
1. Open a TLS connection.
2. Detach both ends of the connection, get raw TCP sockets.
3. If either side reads from TCP is it guaranteed not to receive some leftover TLS data?

**Comment by vlm** [comment]
Oh. This patch (or some modification thereof) is necessary to be able to cleanly close the file descriptors. There's no way to cleanly close TLS without this patch.

The attach-after-detach, or use-tcp-after-dropping-tls is something that has to be tested, but it is a secondary problem.

**Comment by sustrik** [comment]
Ok, merged. Let's keep the cleanup problem in mind though.
